### DISCUSSION TITLE
Error installing sqlcmd in aspnet 8.0 image

### DISCUSSION THREAD

**Discussion by dpalat** [dotnet/dotnet-docker#5261] [Q&A]
# Describe the Bug

I am trying to move to .net 8 from .net 7 but I can't install the sqlcmd package in the image.

I am not sure if a bug or what is happening here.

Dockerfile to reproduce the issue:

```
# Build runtime image
FROM mcr.microsoft.com/dotnet/aspnet:8.0

# install packages
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y \
    curl \
    wget \
    software-properties-common

RUN curl https://packages.microsoft.com/keys/microsoft.asc | tee /etc/apt/trusted.gpg.d/microsoft.asc \
    && add-apt-repository "$(wget -qO- https://packages.microsoft.com/config/debian/11/prod.list)" \
    && apt-get update \
    && apt-get install sqlcmd -y \
    && apt remove software-properties-common -y
```

Error:
ERROR: failed to solve: process "/bin/sh -c apt-get install sqlcmd -y" did not complete successfully: exit code: 100

If you change the line FROM mcr.microsoft.com/dotnet/aspnet:8.0 and use FROM mcr.microsoft.com/dotnet/aspnet:7.0 all works fine.

**Comment by MichaelSimons** [comment]
The version of Debian the .NET images are based on changed between releases.  You can see that in the [tags listing](https://github.com/dotnet/dotnet-docker/blob/main/README.aspnet.md#linux-amd64-tags).

<img width="781" alt="image" src="https://github.com/dotnet/dotnet-docker/assets/8290530/d88b502d-c342-4015-a298-e325797bc4a0">

<img width="739" alt="image" src="https://github.com/dotnet/dotnet-docker/assets/8290530/cfc731ca-144a-4820-b939-bbfbe0c9108f">

Your Dockerfile is registering the Debian/11 package.microsoft.com feed in the .NET 8.0 image if all you are doing is changing from the .NET 7.0 to and 8.0 base image.  Update the debian feed registration and give it a try.

**Comment by dpalat** [comment]
Thanks @MichaelSimons 

I tried with Debian/12 but doesn't work:

```
# Build runtime image
FROM mcr.microsoft.com/dotnet/aspnet:8.0

# install packages
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y \
    curl \
    wget \
    software-properties-common

RUN curl https://packages.microsoft.com/keys/microsoft.asc | tee /etc/apt/trusted.gpg.d/microsoft.asc \
    && add-apt-repository "$(wget -qO- https://packages.microsoft.com/config/debian/12/prod.list)" 
RUN apt-get update \
RUN apt-get install sqlcmd -y \
RUN apt remove software-properties-common -y
```

Error:
0.533 Unable to handle repository shortcut 'deb [arch=amd64,arm64,armhf signed-by=/usr/share/keyrings/microsoft-prod.gpg] https://packages.microsoft.com/debian/12/prod bookworm main'

**Reply by MichaelSimons** [reply]
I don't see a sqlcmd package being published to package.microsoft.com for debian 12.  I would follow-up with the owners on what their support is for debian 12.  This is not an issue with the .NET image itself.
